bin/publish.ml

```diff
@@ -7,25 +7,20 @@
 open Bos_setup
 open Dune_release
 
-let gen_doc ~dry_run ~force dir pkg_name =
-  let build_doc = Cmd.(v "dune" % "build" % "-p" % pkg_name % "@doc") in
-  let doc_dir = Fpath.(v "_build" / "default" / "_doc") in
-  let html_dir = Fpath.(doc_dir / "_html") in
-  let do_doc () =
-    Sos.run ~dry_run ~force build_doc >>= fun () ->
-    Sos.dir_exists ~dry_run html_dir >>| function
-    | true  -> Fpath.(dir // html_dir)
-    | false -> Fpath.(dir // doc_dir)
-  in
-  R.join @@ Sos.with_dir ~dry_run dir do_doc ()
-
-let publish_doc ~dry_run pkg =
+let gen_doc ~dry_run ~force dir pkg_names =
+  let names = String.concat ~sep:"," pkg_names in
+  let build_doc = Cmd.(v "dune" % "build" % "-p" % names % "@doc") in
+  let doc_dir = Fpath.(v "_build" / "default" / "_doc" / "_html") in
+  let do_doc () = Sos.run ~dry_run ~force build_doc in
+  R.join @@ Sos.with_dir ~dry_run dir do_doc () >>= fun () -> Ok doc_dir
```

Comment:
This branch works great for me if I change this function to return `Ok Fpath.(dir // doc_dir)`

Reply:
ha yes you are right :-)